### PR TITLE
[libjpeg-turbo] Fixes cross compilation for armv8

### PR DESCRIPTION
**Pull request by KristianJerpetjon**
Specify library name and version:  libjpeg-turbo/all

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libjpeg-turbo/all/conanfile.py [hunk 1]
```diff
@@ -89,11 +93,16 @@ def build_requirements(self):
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         os.rename(self.name + "-" + self.version, self._source_subfolder)
+        if tools.cross_building(self.settings):
+            tools.replace_in_file(os.path.join(self._source_subfolder,"CMakeLists.txt"),"CMAKE_SYSTEM_PROCESSOR","CMAKE_SYSTEM_PROCESSOR_TARGET")
```
Comment (SpaceIm) [inline recipes/libjpeg-turbo/all/conanfile.py:97]: patch in `build()` instead of `source()` please

File: recipes/libjpeg-turbo/all/conanfile.py [hunk 2]
```diff
@@ -134,9 +143,10 @@ def package(self):
         tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
         tools.rmdir(os.path.join(self.package_folder, "doc"))
         # remove binaries and pdb files
-        for pattern_to_remove in ["cjpeg*", "djpeg*", "jpegtran*", "tjbench*", "wrjpgcom*", "rdjpgcom*", "*.pdb"]:
-            for bin_file in glob.glob(os.path.join(self.package_folder, "bin", pattern_to_remove)):
-                os.remove(bin_file)
+        if not self.options.keep_binaries:
+            for pattern_to_remove in ["cjpeg*", "djpeg*", "jpegtran*", "tjbench*", "wrjpgcom*", "rdjpgcom*", "*.pdb"]:
+                for bin_file in glob.glob(os.path.join(self.package_folder, "bin", pattern_to_remove)):
+                    os.remove(bin_file)
```
Comment (SpaceIm) [inline recipes/libjpeg-turbo/all/conanfile.py:149]: don't keep pdb files.

**Comment by Croydon** [comment]
Doesn't this sound like something that Conan cross-building features should be able to handle? It is certainly impracticable adding a `keep_binaries` to every recipe

File: recipes/libjpeg-turbo/all/conanfile.py (hunk 2 above)
Comment (SSE4) [reply recipes/libjpeg-turbo/all/conanfile.py:149]: yes, this is untreated change

File: recipes/libjpeg-turbo/all/conanfile.py [hunk 3]
```diff
@@ -89,11 +93,16 @@ def build_requirements(self):
...
+        if tools.cross_building(self.settings):
+            tools.replace_in_file(os.path.join(self._source_subfolder,"CMakeLists.txt"),"CMAKE_SYSTEM_PROCESSOR","CMAKE_SYSTEM_PROCESSOR_TARGET")
 
     def _configure_cmake(self):
         if self._cmake:
             return self._cmake
         self._cmake = CMake(self, set_cmake_flags=True)
+        if tools.cross_building(self.settings):
+            if self.settings.arch == "armv8":
+                self._cmake.definitions["CMAKE_SYSTEM_PROCESSOR_TARGET"] = "aarch64"
```
Comment (SSE4) [inline recipes/libjpeg-turbo/all/conanfile.py:105]: not sure why the only armv8 is affected? maybe there is a wider conan issue? /cc @jgsogo

File: recipes/libjpeg-turbo/all/conanfile.py (hunk 2 above)
Comment (jgsogo) [reply recipes/libjpeg-turbo/all/conanfile.py:149]: If the executables are already generated, why removing them from the package? I would say that storage is not an issue nowadays (at least, it isn't for us in ConanCenter). I would remove `keep_binaries` option and keep these binaries in the package.

File: recipes/libjpeg-turbo/all/conanfile.py (hunk 3 above)
Comment (jgsogo) [reply recipes/libjpeg-turbo/all/conanfile.py:105]: I'm with @SpaceIm , I don't know what is CMAKE_SYSTEM_PROCESSOR_TARGET, I cannot find it in CMake docs.

File: recipes/libjpeg-turbo/all/conanfile.py (hunk 2 above)
Comment (madebr) [reply recipes/libjpeg-turbo/all/conanfile.py:149]: @KristianJerpetjon 
We have an issue for discussing including pdbs: https://github.com/conan-io/conan-center-index/issues/1982 
https://github.com/conan-io/conan-center-index/issues/1982#issuecomment-659455118 seems to suggest a conan feature will be added for handling pdb's.

File: recipes/libjpeg-turbo/all/conanfile.py (hunk 3 above)
Comment (prince-chrismc) [reply recipes/libjpeg-turbo/all/conanfile.py:105]: maybe it was supposed to be https://cmake.org/cmake/help/v3.1/variable/CMAKE_SYSTEM_PROCESSOR.html ? 🤔

**Comment by madebr** [comment]
I don't fully understand what you're trying to fix.
Has it something to do with this? https://github.com/libjpeg-turbo/libjpeg-turbo/blob/c3bfbde21d81438b731aeefa69011024e8ae08c2/CMakeLists.txt#L57-L64

What is the value of `CMAKE_SYSTEM_PROCESSOR` that cmake detects when using arm architecture?

Conan has a way to overload some cmake defaults: see https://docs.conan.io/en/latest/reference/build_helpers/cmake.html#definitions
You can add `libjpeg-turbo:CONAN_CMAKE_SYSTEM_PROCESSOR=aarch64` to your profile.
But better would be to create a proper cmake toolchain file and add it to the `CONAN_CMAKE_TOOLCHAIN` environment variable.
See https://docs.conan.io/en/latest/reference/env_vars.html#cmake-related-variables

**Comment by SSE4** [comment]
@KristianJerpetjon I don't think it's a right place to have a fix in the recipe(s). I also don't think it's correct to have it exclusively for amrv8. 
reading docs I see:
```
CMAKE_SYSTEM_PROCESSOR : optional, processor (or hardware) of the
target system. This variable is not used very much except for one
purpose, it is used to load a
CMAKE_SYSTEM_NAME-compiler-CMAKE_SYSTEM_PROCESSOR.cmake file,
which can be used to modify settings like compiler flags etc. for
the target. You probably only have to set this one if you are using
a cross compiler where every target hardware needs special build
settings.
```
so it gives me the impression that **CMAKE_SYSTEM_PROCESSOR** is optional and shouldn't be vital.
we also don't set this variable by default at all: https://github.com/conan-io/conan/blob/6287050a4ca74f47bdcccbb82a66eb749046fa8b/conans/client/build/cmake_flags.py#L224

I'd recommend:
1. open conan issue
2. attach logs so we can understand what exactly fails in your case

I believe correct fix may require setting `CMAKE_SYSTEM_PROCESSOR` and `CMAKE_HOST_SYSTEM_PROCESSOR ` in conan, and having a mapping for all the supported architectures (not just armv8).

**Comment by SSE4** [comment]
created https://github.com/conan-io/conan/issues/8025

**Pull request closed without merging**